servers/sunrpc/server.go

```diff
@@ -31,7 +42,7 @@ func New(l net.Listener) *SunRPC {
 	programsList = []Program{
 		newGfHandshake(),
 		newGfDump(),
-		newGfPortmap(),
+		pmap.NewGfPortmap(),
```

Comment:
cool. In plugins patch, I moved programsList to server/sunrpc/program so that this list can be populated from outside. In this case for pmap changes we need not touch this file.

In pmap package,

    import "github.com/gluster/glusterd2/servers/sunrpc/program"

    func init(){
        program.AddProgram(NewGfPortmap())
    }

`program.AddProgram` adds to programsList. also expose `program.GetProgramList` so that you can use it in this file.

In this file import pmap as 

    _ "github.com/gluster/glusterd2/pmap"

init() func is called when pmap is imported, which adds to programsList